src/dispatch/incident_type/views.py

```diff
@@ -42,11 +46,19 @@ def get_incident_types(
 
 @router.post("/", response_model=IncidentTypeRead)
 def create_incident_type(
-    *, db_session: Session = Depends(get_db), incident_type_in: IncidentTypeCreate
+    *,
+    db_session: Session = Depends(get_db),
+    incident_type_in: IncidentTypeCreate,
+    current_user: DispatchUser = Depends(get_current_user),
 ):
     """
-    Create a new incident_type.
+    Create a new incident type.
     """
+    # We restrict the creation of incident types to admins only
+    if current_user.role != UserRoles.admin:
+        raise HTTPException(
+            status_code=403, detail="You do not have permission to create incident types."
```

Comment:
@kevgliss FYI I used `403` instead of `401` because the latter was triggering re-auth when using basic auth provider.

Reply:
Should we change the other 401s we have to 403s?

Reply:
As a complete outsider I would consider it would good practice to expect the same status codes for the same exception raised by an application. But that's just my two cents ¯\_(ツ)_/¯

Reply:
Yeah, I'll take care of it.